core/store/store.go

```diff
@@ -180,6 +178,19 @@ func newStoreWithDialerAndKeyStore(
 	return store
 }
 
+// InitRootDirectory creates the root directory folder where the keystore lives.
+// It also sets the default file permissions to 0700.
+func InitRootDirectory(config *orm.Config) {
+	var permission os.FileMode = 0700
+	if err := os.MkdirAll(config.RootDir(), os.FileMode(permission)); err != nil {
+		logger.Fatal(fmt.Sprintf("Unable to create project root dir: %+v", err))
```

Comment:
I’d return error here and let the caller do a fatal /. Panic.

Reply:
Yeah good call - I guess we want to require the permissions be set.